tpchgen/src/generators.rs

```diff
@@ -511,18 +534,26 @@ impl<'a> Iterator for PartGeneratorIterator<'a> {
 }
 
 /// Records for the SUPPLIER table.
+///
+/// The Display trait is implemented to format the line item data as a string
+/// in the default TPC-H 'tbl' format.
+///
+/// ```text
+/// 1|Supplier#000000001| N kD4on9OM Ipw3,gf0JBoQDd7tgrzrddZ|17|27-918-335-1736|5755.94|each slyly above the careful|
+/// 2|Supplier#000000002|89eJ5ksX3ImxJQBvxObC,|5|15-679-861-2259|4032.68| slyly bold instructions. idle dependen|
+/// ```
 #[derive(Debug, Clone, PartialEq)]
 pub struct Supplier {
     /// Primary key
     pub s_suppkey: i64,
-    /// Supplier name
-    pub s_name: String,
+    /// Supplier name. Formatted as "Supplier#<n>"
+    pub s_name: i64,
     /// Supplier address
     pub s_address: String,
```

Comment:
There are still some string copies such as this one -- it is due to the fact that the lifetime of the underlying string is tied to the generator (there is a buffer) so I can't return it without the item referring to the iterator

I think what we need to do is postpone the display here as well. I have an idea but I also don't think it will make much of a difference as the Strings are only used now for the much smaller tables.

Reply:
Update: I have another PR to avoid these copies:
- https://github.com/clflushopt/tpchgen-rs/pull/39